ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/PantheonPrivateTransaction.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
+
+import tech.pegasys.ethsigner.core.jsonrpc.EeaSendTransactionJsonParameters;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequestId;
+import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.NonceProvider;
+
+import org.web3j.protocol.eea.crypto.RawPrivateTransaction;
+import org.web3j.utils.Base64String;
+import org.web3j.utils.Restriction;
+
+public class PantheonPrivateTransaction extends PrivateTransaction {
+
+  public static PantheonPrivateTransaction from(
+      final EeaSendTransactionJsonParameters transactionJsonParameters,
+      final NonceProvider nonceProvider,
+      final JsonRpcRequestId id) {
+
+    Base64String privacyId =
```

Comment:
Sorry to change my mind, but I think the style you used in EeaPrivateTransaction to validate the argument with the isEmpty is much easier to read than my suggestion of using orElseThrow. Oh and privacyId can be made final.

Reply:
done